src/main/java/org/stellar/sdk/Predicate.java

```diff
@@ -34,9 +37,9 @@ public static Predicate fromXdr(org.stellar.sdk.xdr.ClaimPredicate xdr) {
       case CLAIM_PREDICATE_NOT:
         return new Not(fromXdr(xdr.getNotPredicate()));
       case CLAIM_PREDICATE_BEFORE_RELATIVE_TIME:
-        return new RelBefore(xdr.getRelBefore().getInt64());
+        return new RelBefore(xdr.getRelBefore().getDuration().getInt64());
```

Comment:
I think we should update these predicates to take their respective data types, i.e. `xdr.Duration` instead of the native `long`. Wdyt?

Reply:
I wove that in to reference the xdr temporal types, I noted minor backward compat/breaking aspect of this change in changelog.md. [5bf518bfd24](https://github.com/stellar/java-stellar-sdk/pull/413/commits/5bf518bfd24895520b5e4f391ad71470eda584af)

Reply:
@Shaptic , i backed off a bit on this change and kept the backwards compliant interface for Predicate also - [163b582c5d](https://github.com/stellar/java-stellar-sdk/pull/413/commits/163b582c5d7c1ba9cb11ff69ca626c517af7c1be)

Reply:
good call - I think I didn't realize this was a user-facing abstraction layer so maintaining bwd compatibility makes sense